src/Select.tsx

```diff
@@ -10,20 +10,38 @@ export type SelectProps = Omit<
 
 const StyledSelect = styled.select`
   appearance: none;
-  background-color: transparent;
   border: 0;
   color: currentColor;
+  font-size: inherit;
   outline: none;
   width: 100%;
 
-  option {
-    color: initial;
-  }
-
   /* colors the select input's placeholder text */
-  &:invalid {
+  &[data-hasplaceholder='true'] {
     color: ${get('colors.fg.subtle')};
   }
+
+  /* Firefox hacks: */
+  /* 1. Reverts color of non-placeholder options in the dropdown */
+  &[data-hasplaceholder='true'] option:not(:first-child):not(:disabled),
+  optgroup:not(:disabled) {
+    color: ${get('colors.fg.default')};
+  }
+
+  /* 2. Makes Firefox's native dropdown menu's background match the theme.
+
+        background-color should be 'transparent', but Firefox uses the background-color on 
+        <select> to determine the background color used for the dropdown menu.
+  */
+  background-color: inherit;
+
+  /* 3. Prevents visible overlap of partially transparent background colors.
+  
+     'colors.input.disabledBg' happens to be partially transparent in light mode, so we use a
+     transparent background-color on a disabled <select>. */
+  &:disabled {
+    background-color: transparent;
+  }
```

Comment:
Help wanted: I don't like this solution, but I've yet to come up with anything better.